src/instructlab/train/pytorch_train.py

```diff
@@ -193,40 +201,45 @@ def linux_train(
     )
 
     if four_bit_quant:
-        print("LINUX_TRAIN.PY: USING 4-bit quantization with BitsAndBytes")
-        use_fp16 = True
+        print("PYTORCH_TRAIN.PY: USING 4-bit quantization with BitsAndBytes")
         bnb_config = BitsAndBytesConfig(
             load_in_4bit=True,
             bnb_4bit_quant_type="nf4",
             bnb_4bit_use_double_quant=True,
             bnb_4bit_compute_dtype=torch.float16,  # if not set will throw a warning about slow speeds when training
         )
     else:
-        print("LINUX_TRAIN.PY: NOT USING 4-bit quantization")
-        use_fp16 = False
+        print("PYTORCH_TRAIN.PY: NOT USING 4-bit quantization")
         bnb_config = None
 
     # Loading the model
-    print("LINUX_TRAIN.PY: LOADING THE BASE MODEL")
+    print("PYTORCH_TRAIN.PY: LOADING THE BASE MODEL")
     config = AutoConfig.from_pretrained(
         model_name, torchscript=True, trust_remote_code=True
     )
 
     model = AutoModelForCausalLM.from_pretrained(
         model_name,
-        torch_dtype="auto",
+        torch_dtype=dtype,
         quantization_config=bnb_config,
         config=config,
         trust_remote_code=True,
         low_cpu_mem_usage=True,
     )
+    #  if model.dtype == torch.bfloat16:
+    # bfloat16 slows down training... alot.
+    #      model.dtype = torch.float16
```

Comment:
Same comment about commented code - needed? or can we can remove?

Reply:
removed